ballerina/soap_utils.bal

```diff
@@ -15,94 +15,32 @@
 // under the License.
 
 import ballerina/http;
-import ballerina/lang.'xml as xmllib;
 import ballerina/mime;
 
-# Provides an empty SOAP envelope for the given SOAP version.
-#
-# + soapVersion - The SOAP version of the request
-# + return - XML with the empty SOAP envelope
-function createSoapEnvelop(SoapVersion soapVersion) returns xmllib:Element {
-    if soapVersion == SOAP11 {
-        return <xmllib:Element> xml `<soap:Envelope
-        xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/"
-        soap:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/">
-                </soap:Envelope>`;
-    } else {
-        return <xmllib:Element> xml `<soap:Envelope
-        xmlns:soap="http://www.w3.org/2003/05/soap-envelope"
-        soap:encodingStyle="http://www.w3.org/2003/05/soap-encoding">
-                </soap:Envelope>`;
-    }
-}
-
-# Provides the SOAP headers in the request as XML.
-#
-# + soapVersion - The SOAP version of the request
-# + return - XML with the empty SOAP header
-function createSoapHeader(SoapVersion soapVersion) returns xml {
-    xmllib:Element headersRoot;
-    if soapVersion == SOAP11 {
-        headersRoot = <xmllib:Element> xml `<soap:Header xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/"></soap:Header>`;
-    } else {
-        headersRoot = <xmllib:Element> xml `<soap:Header xmlns:soap="http://www.w3.org/2003/05/soap-envelope"></soap:Header>`;
-    }
-    return headersRoot;
-}
-
-# Provides the SOAP body in the request as XML.
-#
-# + payload - The payload to be sent
-# + soapVersion - The SOAP version of the request
-# + return - XML with the SOAP body
-function createSoapBody(xml payload, SoapVersion soapVersion) returns xml {
-    xmllib:Element bodyRoot;
-    if soapVersion == SOAP11 {
-        bodyRoot = <xmllib:Element> xml `<soap:Body xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/"></soap:Body>`;
-    } else {
-        bodyRoot = <xmllib:Element> xml `<soap:Body xmlns:soap="http://www.w3.org/2003/05/soap-envelope"></soap:Body>`;
-    }
-    bodyRoot.setChildren(payload);
-    return bodyRoot;
-}
-
 # Prepares a SOAP envelope with the XML to be sent.
 #
 # + soapAction - SOAP action
 # + body - SOAP request body as an `XML` or `mime:Entity[]` to work with soap attachments
 # + soapVersion - The SOAP version of the request
 # + return - The SOAP Request sent as `http:Request` with the SOAP envelope
-function fillSoapEnvelope(SoapVersion soapVersion, xml | mime:Entity[] body, string? soapAction = ())
+function fillSoapEnvelope(SoapVersion soapVersion, xml|mime:Entity[] body, string soapAction)
```

Comment:
Shall we rename this function to a meaningful name like `createHttpRequest` or `updateHttpRequest` etc ?

Reply:
Fixed in [6d320c2](https://github.com/ballerina-platform/module-ballerina-soap/pull/93/commits/6d320c2b63f1532033f3a323bb3cf79396b125d5)